packages/calcite-components/src/components/sheet/sheet.tsx

```diff
@@ -224,6 +225,26 @@ export class Sheet implements OpenCloseComponent, FocusTrapComponent, LoadableCo
     this.handleMutationObserver(),
   );
 
+  //--------------------------------------------------------------------------
+  //
+  //  Event Listeners
+  //
+  //--------------------------------------------------------------------------
+
+  @Listen("keydown", { target: "window" })
```

Comment:
Instead of creating this event on the window, can we add it on the host of the component? That way, it won't close for an escape that happens outside of the component.

Reply:
Same thing with the modal component. Although less important since the modal is deprecated :D

Reply:
Oh, I initially had it on host but @jcfranco thinks we should close the modal regardless of where esc occurs.

Reply:
Ok. However, when we refactor sheet to support #10364, this would need to move to the host when non modal. It might be better to just have it on sheets host for now. I'll defer to @jcfranco

Reply:
When a sheet can be non modal, we wouldn't want escape closing it from outside of the sheet.